include/listener.h

```diff
@@ -43,7 +43,7 @@ class Listener {
 
     Listener(const Address& address);
     void init(
-            size_t workers,
+            ssize_t workers,
```

Comment:
bacause the value will be compared somewhere and the compile warns that you compare signed value with an unsigned one. e.g. hardware_concurrency() returns init which is a signed value.

Also you could make all size_t, but you should decide which one to use and stick to it